test/unit/HistoricalSubscription.test.js

```diff
@@ -0,0 +1,550 @@
+                })
+
+                describe('when message verification returns false', () => {
+                    it('does not call the message handler', async () => sub.handleResentMessage(msg, sinon.stub().resolves(false)))
+
+                    it('prints to standard error stream', async () => {
+                        await sub.handleResentMessage(msg, sinon.stub().resolves(false))
+                        assert(console.error.calledWith(sinon.match.instanceOf(InvalidSignatureError)))
+                    })
+
+                    it('emits an error event', async (done) => {
+                        sub.on('error', (err) => {
+                            assert(err instanceof InvalidSignatureError)
+                            done()
+                        })
+                        return sub.handleResentMessage(msg, sinon.stub().resolves(false))
+                    })
+                })
+
+                describe('when message verification throws', () => {
+                    it('emits an error event', async (done) => {
+                        const error = new Error('test error')
+                        sub.on('error', (err) => {
+                            assert(err instanceof VerificationFailedError)
+                            assert.strictEqual(err.cause, error)
+                            done()
+                        })
+                        return sub.handleResentMessage(msg, sinon.stub().throws(error))
+                    })
+                })
+
+                describe('when message handler throws', () => {
+                    it('emits an error event', async (done) => {
+                        sub.on('error', (err) => {
+                            assert.strictEqual(err.name, 'should not be called!')
+                            done()
+                        })
+                        return sub.handleResentMessage(msg, sinon.stub().resolves(true))
+                    })
+
+                    it('prints to standard error stream', async (done) => {
+                        sub.on('error', (err) => {
+                            assert.strictEqual(err.name, 'should not be called!')
+                            done()
+                        })
+                        return sub.handleResentMessage(msg, sinon.stub().resolves(true))
+                    })
+                })
+            })
+
+            it('calls the callback once for each message in order', () => {
+                const msgs = [1, 2, 3, 4, 5].map((timestamp) => createMsg(
+                    timestamp,
+                    timestamp === 1 ? 0 : timestamp - 1,
+                ))
+
+                const received = []
+
+                const sub = new HistoricalSubscription(msg.getStreamId(), msg.getStreamPartition(), (content, receivedMsg) => {
+                    received.push(receivedMsg)
+                    if (received.length === 5) {
+                        assert.deepEqual(msgs, received)
```

Comment:
I would add a `done()` in here to be extra sure that the callbacks (and hence this assertion) were indeed invoked.